chain/client/src/stateless_validation/chunk_validator/mod.rs

```diff
@@ -548,6 +578,27 @@ pub(crate) fn validate_chunk_state_witness(
         )));
     }
 
+    // Compute receipt hashes here to avoid copying receipts
+    let outgoing_receipts_hashes = {
+        let shard_layout = epoch_manager
+            .get_shard_layout_from_prev_block(state_witness.chunk_header.prev_block_hash())?;
+        Chain::build_receipts_hashes(&outgoing_receipts, &shard_layout)
+    };
+    // Save main state transition result to cache.
+    {
+        let mut shard_cache = main_state_transition_cache.write().unwrap();
+        let cache = shard_cache
+            .entry(shard_uid)
+            .or_insert_with(|| LruCache::new(NUM_WITNESS_RESULT_CACHE_ENTRIES));
+        cache.put(
+            block_hash,
```

Comment:
which block is this the one with height_created? otherwise if we miss the chunk, won't the block 
to include the chunk change?

Reply:
No this is the one with `height_included`, i.e, the block whose processing leads to the chunk being applied.